packages/manager/apps/cloud/client/app/deskaas/deskaas-get-console-access/deskaas-get-console-access.controller.js

```diff
@@ -1,17 +1,43 @@
+import get from 'lodash/get';
 
+export default class DeskaasGetConsoleAccessCtrl {
+  /* @ngInject */
+  constructor($translate, OvhApiDeskaasService) {
+    this.$translate = $translate;
+    this.OvhApiDeskaasService = OvhApiDeskaasService;
+  }
 
-angular.module('managerApp')
-  .controller('DeskaasGetConsoleAccessCtrl',
-    function DeskaasGetConsoleAccessCtrl($scope, $location, $uibModalInstance) {
-      const self = this;
+  ok() {
+    // Call POST /console to create the task, an email will be sent to the user
+    return this.getConsole();
+  }
 
-      self.cancel = function cancel() {
-        // Remove popup
-        $uibModalInstance.dismiss('cancel');
-      };
+  cancel() {
+    // Remove popup
+    this.goBackToDetails();
+  }
 
-      self.ok = function ok() {
-        // Call POST /console to create the task, an email will be sent to the user
-        $uibModalInstance.close(self.values);
-      };
-    });
+  getConsole() {
+    const promise = this.OvhApiDeskaasService.v6()
+      .console({ serviceName: this.serviceName }, null).$promise;
+
+    return promise.then(result => this.goBackToDetails(
+      this.$translate.instant('vdi_console_task', {
+        serviceName: this.serviceName,
+      }),
+      'success',
+      {
+        task: result.taskId,
+      },
+    ))
+      .catch((error) => {
```

Comment:
```suggestion
      .catch((error) => 
```

you can chain the promises this way 🙂